src/collect/cli.rs

```diff
@@ -95,6 +95,18 @@ Examples of meta filters:
 --filter-meta 'sk_buff.dev.nd_net.net.ns.inum == 4026531840'"#
     )]
     pub(super) meta_filter: Option<String>,
+    #[arg(
+        short,
+        long,
+        default_value = "false",
+        help = "Auto-probe mode evaluates where Retis could add additional probes based on functions reported
+in the events stack traces ((their display is still controlled by --stack). All matching functions are probed
+at runtime using kprobes.
+
+Using a filter is required. If no explicit probe is given, skb:kfree_skb and skb:consume_skb
```

Comment:
We cannot "track-back", meaning, the filter will match the packet as it is in the [automatic] probes set (e.g. `kfree_skb`, `consume_skb` or manually, e.g. in the tx path using `dev_hard_start_xmit` and so forth).

The filter then is also set globally, which means that the packet matched in `dev_hard_start_xmit` may not be matched before nat happens, assuming nat (same thing for the rx path).
If confirmed, should we "call out" this behavior?

Reply:
You're right; what about:

```rust
    #[arg(
        long,
        default_value = "false",
        help = "When set, evaluates where Retis could add additional probes based on functions reported
in the events stack traces (their display is still controlled by --stack). All matching
functions are probed at runtime using kprobes.

Notes:
- Using a filter is required.
- If no explicit probe is given, tp:skb:kfree_skb and tp:skb:consume_skb are used as a
  starting point.
- Additional probes are added only after events including them in their stack trace are
  reported; this means the first packets hitting a probe won't be reported.
- Packets will only be followed prior to the initial set of probes (as this mode uses
  stack traces). This also means the filter must match packets as they appear in the
  initial set of probes; packet transformation can't be automatically detected."
```

Reply:
yes, sgtm